autoneg.go

```diff
@@ -76,62 +73,71 @@ func (accept accept_slice) Less(i, j int) bool {
 	return false
 }
 
-func (accept accept_slice) Swap(i, j int) {
-	slice := []Accept(accept)
+func (slice AcceptSlice) Swap(i, j int) {
 	slice[i], slice[j] = slice[j], slice[i]
 }
 
+// trimSpaces trims spaces from the given string.
+func trimSpaces(s string) string {
+	if len(s) == 0 || (s[0] != ' ' && s[len(s)-1] != ' ') {
+		// strings.Trim performs memory allocations, so implement
+		// fast path to avoid those if not needed.
+		return s
+	}
+	return strings.Trim(s, " ")
+}
+
 // Parse an Accept Header string returning a sorted list
 // of clauses
-func ParseAccept(header string) (accept []Accept) {
+func ParseAccept(header string) (accept AcceptSlice) {
```

Comment:
Does it make sense to expose `AcceptSlice` to consumers of this package given that `ParseAccept` already sorts the slice anyway? Can this be kept as `[]Accept` to avoid leaking the implementation details to the consumer?

Reply:
SGTM - fixed